### PR TITLE
Convert Janus metadata JSON into typed object hierarchy

### PR DESCRIPTION
**Pull request by rjkip**
[Pivotal Tracker #104886410](https://www.pivotaltracker.com/story/show/104886410)

### CODE REVIEW

**Comment by rjkip** [comment]
@DRvanR Any thoughts before I write a wee set of tests?

File: src/Surfnet/Conext/EntityVerificationFramework/Value/PemEncodedX509Certificate.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    /**
+     * @param string $data
+     * @param string $propertyPath
+     * @return PemEncodedX509Certificate
+     */
+    public static function deserialise($data, $propertyPath)
+    {
+        Assert::string($data, 'Certificate data must be a string', $propertyPath);
+
+        $certificate = new PemEncodedX509Certificate();
```
Comment (DRvanR) [inline src/Surfnet/Conext/EntityVerificationFramework/Value/PemEncodedX509Certificate.php]: is there a reason for mixing `new ClassName()` and `new self()`? Not that I mind, just wondering.

**Comment by rjkip** [comment]
:bell: @arothuis

**Comment by DRvanR** [comment]
:+1:

File: src/Surfnet/Conext/EntityVerificationFramework/Value/ConfiguredMetadata.php [hunk 2]
```diff
@@ -18,6 +18,221 @@
...
+    /** @var mixed[] Array indexed by string keys */
+    private $freeformProperties = [];
+
+    /**
+     * @param mixed $data
+     * @return ConfiguredMetadata
+     */
+    public static function deserialise($data)
+    {
+        Assert::isArray($data, 'Configure metadata data must be an array structure');
```
Comment (arothuis) [inline src/Surfnet/Conext/EntityVerificationFramework/Value/ConfiguredMetadata.php]: Nit: missing 'd'.

**Comment by arothuis** [comment]
ConfiguredMetadata's a bit of a large class, but that's understandable given its nature :+1:

**Pull request merged**